pkg/storage/tsdb/users/updater.go

```diff
@@ -0,0 +1,36 @@
+package users
+
+import (
+	"context"
+	"time"
+
+	"github.com/thanos-io/objstore"
+)
+
+type UserIndexUpdater struct {
+	bkt     objstore.InstrumentedBucket
+	scanner Scanner
+}
+
+func NewUserIndexUpdater(bkt objstore.InstrumentedBucket, scanner Scanner) *UserIndexUpdater {
+	return &UserIndexUpdater{
+		bkt:     bkt,
+		scanner: scanner,
+	}
+}
+
+func (u *UserIndexUpdater) UpdateUserIndex(ctx context.Context) error {
```

Comment:
Would be nice to have a metric on this. So we can track when it stopped updating. Similar to bucketIndex for tenant